src/ai/backend/client/func/service.py

```diff
@@ -99,19 +99,19 @@ async def create(
         model_mount_destination: Optional[str] = None,
         envs: Optional[Mapping[str, str]] = None,
         startup_command: Optional[str] = None,
-        resources: Optional[Mapping[str, str | int]] = None,
-        resource_opts: Optional[Mapping[str, str | int]] = None,
+        resources: Mapping[str, str | int],
+        resource_opts: Mapping[str, str | int],
         cluster_size: int = 1,
         cluster_mode: Literal["single-node", "multi-node"] = "single-node",
-        domain_name: Optional[str] = None,
-        group_name: Optional[str] = None,
+        domain_name: str,
+        group_name: str,
         bootstrap_script: Optional[str] = None,
         tag: Optional[str] = None,
         architecture: Optional[str] = DEFAULT_IMAGE_ARCH,
-        scaling_group: Optional[str] = None,
+        scaling_group: str,
```

Comment:
Isn’t it the case that once a parameter is given a default value, all following parameters must also have default values? @rapsealk

Reply:
Any parameter defined after `*` must be specified using its name (i.e., as a keyword argument). The order of these keyword-only arguments in a function call doesn't matter, since they can't be passed positionally.

Reply:
Wouldn’t it be better to place keyword-only arguments without default values first for better readability and clarity. @rapsealk